lib/pattern-importer.js

```diff
@@ -0,0 +1,155 @@
+var through = require('through2'),
+  utility = require('./utility'),
+  merge = require('lodash.merge'),
+  gutil = require('gulp-util'),
+  path = require('path'),
+  fs = require('fs'),
+  PluginError = gutil.PluginError,
+  PLUGIN_NAME = 'pattern-importer';
+
+/*
+ * Imports html patterns that are following the standards at github.com/pattern-library
+ *
+ * @param {Object} options
+*/
+exports.patternImporter = function (options) {
```

Comment:
@scottnath I was able to unit test this by adding `convertCategoryTitlesDataFile: '../lib/data/pattern-lab-categories.yml',` to the `options` object in test/main.js. It runs ok. I see that this sets a series of operations to perform on a stream wherever this type of stream may be instantiated.